k8s/operators/src/mayastorpool/crd.rs

```diff
@@ -0,0 +1,76 @@
+use kube::CustomResource;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+#[derive(CustomResource, Serialize, Deserialize, Default, Debug, PartialEq, Clone, JsonSchema)]
+#[kube(
+group = "openebs.io",
+version = "v1alpha1",
+kind = "MayastorPool",
+plural = "mayastorpools",
+// The name of the struct that gets created that represents a resource
+namespaced,
+status = "MayastorPoolStatus",
+derive = "PartialEq",
+derive = "Default",
+shortname = "msp",
+printcolumn = r#"{ "name":"node", "type":"string", "description":"node the pool is on", "jsonPath":".spec.node"}"#,
+printcolumn = r#"{ "name":"status", "type":"string", "description":"pool status", "jsonPath":".status.state"}"#,
+printcolumn = r#"{ "name":"capacity", "type":"integer", "format": "int64", "minimum" : "0", "description":"total bytes", "jsonPath":".status.capacity"}"#,
+printcolumn = r#"{ "name":"used", "type":"integer", "format": "int64", "minimum" : "0", "description":"used bytes", "jsonPath":".status.used"}"#,
+printcolumn = r#"{ "name":"available", "type":"integer", "format": "int64", "minimum" : "0", "description":"available bytes", "jsonPath":".status.available"}"#
+)]
+
+/// The pool spec which contains the parameters we use when creating the pool.
+pub struct MayastorPoolSpec {
```

Comment:
do these need to be pub or is pub(crate) enough? We don't need to expose these to the library?